pandas/tests/indexing/multiindex/test_multiindex.py

```diff
@@ -206,3 +206,22 @@ def test_multiindex_with_na_missing_key(self):
         )
         with pytest.raises(KeyError, match="missing_key"):
             df[[("missing_key",)]]
+
+    def test_multiindex_dtype_preservation(self):
+        # GH51261
+        df = DataFrame(
+            ["value"],
+            columns=MultiIndex.from_tuples([("A", "B")], names=["lvl1", "lvl2"]),
+        ).astype("category")
```

Comment:
```suggestion
        columns = MultiIndex.from_tuples([("A", "B")], names=["lvl1", "lvl2"])
        df = DataFrame(["value"], columns=columns).astype("category")
```

To make it a bit easier to read